msal/cache/cache.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/*
+Package cache allows third parties to implement external storage for caching token data
+for distributed systems or multiple local applications access.
+
+The data stored and extracted will represent the entire cache. Therefore it is recommended
+one msal instance per user. This data is considered opaque and there are no guarantees to
+implementers on the format being passed.
+*/
+package cache
+
+// Marshaler marshals data from an internal cache to bytes that can be stored.
+type Marshaler interface {
+	Marshal() ([]byte, error)
+}
+
+// Unmarshaler unmarshals data from a storage medium into the internal cache, overwriting it.
+type Unmarshaler interface {
+	Unmarshal([]byte) error
+}
+
+// Serializer can seralize the cache to binary or from binary into the cache.
+type Serializer interface {
+	Marshaler
+	Unmarshaler
+}
+
+// Token is used to provide external storage of token data. The data being passed is considered
+// opaque.
+type Token interface {
```

Comment:
Token also doesn't seem quite right here. Token, in the context of MSAL has a clear meaning - A JWT with claims used for authentication/authorization. It also doesn't have a "er" suffix, which I I think is Go idiom and which is used elsewhere. 

This type is something completely unrelated to the above definition - is allows a user to set callbacks for marshalling/unmarshalling of the in-memory token cache. I can't think of a better name myself, but if I was to choose the previous name "accessor" seems more appropriate

Reply:
As to the Go idiom: "er" is only added to interface types that have a single method defined.  So instead of having Read.Read(), you have Reader.Read().  This does not apply on multi-method interfaces.

Do you feel the same way knowing that it is always:  cache.Token?   
We could just go all the way:  cache.ReadReplace